### PR TITLE
Porting API-Samples to Android.

### PR DESCRIPTION
**Pull request by hak**
- Add API-Samples/android folder for AndroidStudio template files.
- Update CMakeLists.txt to support AndroidStudio project generation.
To generate AS project:
cmake -DANDROID=ON -DANDROID_ABI=[armeabi-v7a|arm64-v8a|x86|x86_64|all(default)]
and import API-Samples/android in AndroidStudio.
- Add Android support in API-Samples/utils.
- Update some sample code if util functions couldn't abstract platform dependencies.

### CODE REVIEW

**Comment by Tony-LunarG** [comment]
The strategy for LICENSE.txt and LICENSE_SAMPLES.txt is that because VulkanSamples is downstream from LoaderAndValidationLayers, LICENSE.txt is the unaltered license set from LVL.  LICENSE_SAMPLES.txt are additional licenses applicable only to the VS repo.  We did it to try to avoid merge conflicts when updating the LVL content.  What is the legal objection?

File: API-Samples/utils/util_init.cpp [hunk 1]
```diff
@@ -415,7 +427,9 @@ void destroy_debug_report_callback(struct sample_info &info) {
 }
 
 void init_connection(struct sample_info &info) {
-#ifndef _WIN32
+#ifdef __ANDROID__
+    // Do nothing on Android.
+#elif defined(_WIN32)
```
Comment (Tony-LunarG) [inline API-Samples/utils/util_init.cpp]: I  think this is reversed and should be not defined

Comment (hak) [inline API-Samples/utils/util_init.cpp]: Done

Comment (hak) [inline API-Samples/utils/util_init.cpp]: His objection was at the point that 2 files (LICENSE.txt and LICENSE) are vague to distinguish and  now it looks better.
I would withdraw the change for an instance and will update if he still has a suggestion.

**Comment by Tony-LunarG** [comment]
Committed to master - thanks!  And thanks for the second submission - much cleaner than the first.

**Pull request closed without merging**